test/sparoid_test.rb

```diff
@@ -39,6 +39,19 @@ def test_it_sends_message
     end
   end
 
+  def test_it_uses_public_ip_override
+    key = "0000000000000000000000000000000000000000000000000000000000000000"
+    hmac_key = "0000000000000000000000000000000000000000000000000000000000000000"
+    UDPSocket.open do |server|
+      server.bind("127.0.0.1", 0)
+      port = server.addr[1]
+      s = Sparoid::Instance.new
+      s.stub(:public_ip, ->(*_) { raise "public_ip method not expected to be called" }) do
+        s.auth(key, hmac_key, "127.0.0.1", port, public_ip: "127.0.1.1")
```

Comment:
Is this the intention of this test?

```suggestion
      s.stub(:cached_public_ip, ->(*_) { raise "cached_public_ip method not expected to be called" }) do
        s.auth(key, hmac_key, "127.0.0.1", port, public_ip: "127.0.1.1")
```

Reply:
Ah! I didn't read the code carefully, my confusion lied within the argument `public_ip`, didn't spot we had a method with the same name too. Think that is another point for us trying to find another name for the argument? 😄